doc/generate_cet.py

```diff
@@ -3,7 +3,144 @@
+                    gen_topic_name = gen_topic.find("EFDB_Topic")
+                    gen_topic_short_name_array = gen_topic_name.text.split("_")[1:]
+                    gen_topic_short_name = '_'.join(gen_topic_short_name_array)
+                    if gen_topic_short_name in has_generics:
+                        topic_name = gen_topic.find('EFDB_Topic').text
+                        short_name = topic_name.split("_")[-1]
+                        cf.write(f"{short_name}\n")
+                        cf.write(f"{'~'*len(short_name)}\n")
+                        gen_topic_description = gen_topic.find("Description")
+                        if gen_topic_description is not None:
+                            cf.write("**Description**: ")
+                            cf.write(f"{gen_topic_description.text}\n")
+                            cf.write(f"\n")
+                        for gen_field in gen_topic:
+                            if gen_field.tag == "item":
+                                gen_field_name = gen_field.find('EFDB_Name')
+                                gen_field_description = gen_field.find('Description')
+                                cf.write("\n")
+                                cf.write(f".. _{subsystem}:{short_name}:{gen_field_name.text}:\n\n")
+                                cf.write(f"{gen_field_name.text}\n")
+                                cf.write(f"{'*'*len(gen_field_name.text)}\n")
+                                for gen_attribute in gen_field:
+                                    if gen_attribute.tag in ["Count", "IDL_Size"]:
+                                        if gen_attribute.text == "1":
+                                            pass
+                                        else:
+                                            cf.write(f":{gen_attribute.tag}: {gen_attribute.text}\n")
+                                    elif gen_attribute.tag in ignored_attributes:
+                                        pass
+                                    else:
+                                        cf.write(f":{gen_attribute.tag}: {gen_attribute.text}\n")
+                                cf.write("\n")
+                                cf.write("**Description**: ")
+                                cf.write(f"{gen_field_description.text}\n")
+                                cf.write("\n")
+                            elif gen_field.tag in ignored_fields:
+                                pass
+                            else:
+                                cf.write(f":{gen_field.tag}: {gen_field.text}\n")
+                            cf.write("\n")
+                    else:
+                        pass
+
 def main():
+    """Generates CSC rst documentation from XML.
+
+    Currently goes through each topic file within each CSC folder.
+    Alphabetizing is a little weird because CSC specific topics and generic topics are located in separate
+    xml files and so each file is alphabetized seprately.
```

Comment:
seprately -> separately

I finally get it. You list specific topics first, then generic topics. Please consider stating that a bit more explicitly, something along the lines of "Component-specific topics are listed first, then generic topics. Topics are alphabetized within each group."